src/ngraph/CMakeLists.txt

```diff
@@ -521,6 +521,14 @@ if(NOT NGRAPH_JSON_ENABLE)
     target_compile_definitions(ngraph PUBLIC NGRAPH_JSON_DISABLE)
 endif()
 
+if(NGRAPH_INTERPRETER_STATIC_LIB_ENABLE)
```

Comment:
If you are disabling the ability for a static ngraph to load dynamic libraries then you don't need the option to build a backend as a shared library. That makes no sense. You would builld a library that you can't use. 
There should be only one flag, `NGRAPH_STATIC_LIB_ENABLE` and if you try to build a backend which does not support static linkage it should report a cmake error. Remove these individual backend enable flags.